public/scripts/tagsList.js

```diff
@@ -1,21 +1,148 @@
-function getTagObject(content){
+let labelArray = null
+let currentFocus = -1
+
+function findLabelSuggestion(input) {
+    if (input === '')
+        return []
+    let suggestion = []
+
+    suggestion = labelArray.filter(element => {
+        if (input.length > element.length)
+            return false
+        return input.toUpperCase() === element.substring(0, input.length).toUpperCase()
+    })
+
+    return suggestion
+}
+
+function createTagObject(content) {
     let template = document.createElement('template')
-    template.innerHTML = 
-    `<div class="tag-element">
+    template.innerHTML =
+        `<div class="tag-element">
         <span class="tag-element-text" data-value=${content}>${content}</span> 
-        <button class="delete-tag-button"></button> 
+        <button class="delete-tag-button" type="button">x</button> 
     </div>`
+    template.content.querySelector('button').onclick = (event) => {
+        event.target.parentNode.remove()
+    }
     return template.content  // this is so much simple :O
+}
+
+function closeRecommendationList() {
+    let lists = document.querySelectorAll('.autocomplete-list')
+    if (lists) {
+        lists.forEach(element => element.remove())
+    }
+}
+
+function setActive(autocompleteList) {
+    try {
+        setAllInactive()
+        autocompleteList.querySelectorAll('.unit-suggestion')[currentFocus].classList.add('autocomplete-active')
+    } catch (error) {
+        console.log(error)
+    }
 
 }
-(function(){
-    var tagInput = document.getElementById("tag-input")
-    var tagList = document.getElementById("tag-list")
-    tagInput.oninput =  () => {
-        // we will calculate the search options
+
+function setAllInactive() {
+    let autocompleteList = document.getElementById('autocomplete-list')
+    autocompleteList.querySelectorAll('.unit-suggestion').forEach(element => {
+        element.classList.remove('autocomplete-active');
+    })
+}
+
+function displayAutocompleteOptions(tagInput, suggestion) {
+    let template = document.createElement('template')
+    let searchValue = tagInput.value
+    let autocompleteDiv = document.createElement('div')
+    autocompleteDiv.setAttribute('class', 'autocomplete-list')
+    autocompleteDiv.setAttribute('id', 'autocomplete-list')
+
+    suggestion.forEach((element) => {
```

Comment:
`suggestion.forEach`... can be misleading. Shouldn't that be an array, and - if so - shouldn't this be plural? This will also suggest what the `displayAutocompleteOptions` functions asks for. In your case, the second argument could well be a string, or an object, but not necessarily an array/a list of something. So perhaps `autocompleteList` instead of `suggestion`?